template/domain/src/main/java/co/nimblehq/template/domain/usecase/UseCase.kt

```diff
@@ -2,16 +2,12 @@ package co.nimblehq.template.domain.usecase
 
 import co.nimblehq.template.domain.model.Model
 import co.nimblehq.template.domain.repository.Repository
+import kotlinx.coroutines.flow.Flow
 import javax.inject.Inject
 
 class UseCase @Inject constructor(private val repository: Repository) {
 
-    suspend fun execute(): UseCaseResult<List<Model>> {
-        return try {
-            val response = repository.getModels()
-            UseCaseResult.Success(response)
-        } catch (e: IllegalStateException) {
-            UseCaseResult.Error(e)
-        }
+    fun execute(): Flow<List<Model>> {
+        return repository.getModels()
```

Comment:
@luongvo Hmm, Do we still need `UseCase` if we don't handle any of the results(`success/error`) in `UseCase`? I worried that it would have a lot of logic inside `VM` instead 🤔

Reply:
@Wadeewee From what we can see from VM implementation, we have `catch` and `collect` to subscribe for success or error case already. We can leave the usecase back for some data preparation (request) or data consolidation (response) actions. 🤔

Reply:
> From what we can see from VM implementation, we have catch and collect to subscribe for success or error case already. It's very clear.

I understand this point: ☝️ @luongvo, but my concern is when handling the chain calls because the previous way with `UseCaseResult` in `UseCase`, we can inject another `UseCase,` So whenever it's `success/error` and we need to call something from another UseCase to get the result, Let's say we want to call some `API` when the error occurs So, by this way we can execute the next `UseCase` in current UseCase instead of filter in `VM` 🙏  It seems like using `UseCaseResult` it's more flexible, IMO🤔

Reply:
> So, by this way we can execute the next UseCase in current UseCase instead of filter in VM 🙏 It seems like using UseCaseResult it's more flexible, IMO🤔

@Wadeewee I think you're missing and remember how `flatMap` benefits us when working with RxJava. So switching to Flow, we have something called `flatMapConcat` to map multiple flows in a chain. Please checkout more sample here https://github.com/nimblehq/android-templates/issues/213#issuecomment-1135375842

Do you think which one is better, the left or the right side? 🤓 

![image](https://user-images.githubusercontent.com/16315358/187593374-91f034f2-5499-4304-ba35-2b33d162d3bd.png)

Reply:
> Do you think which one is better, the left or the right side? 🤓

@luongvo If we handle the `Chain` call in `VM` like your example.  Surely, the right side without `UseCaseResult` is better, but Actually, we can handle it in `UseCase` instead of `VM,` So in `VM,` we will handle it only 
- if success -> emit result value to view or do something but not `execute another usecase.`
- If error -> emit error to view or do something but not `execute another usecase.` 

Currently, our template doesn't have complex logic, So I think it's good to go without `UseCaseResult,` but TBH, I'm not sure for the `Client` project that might have complex logic.

Reply:
@Wadeewee A usecase should do a single job only and it should not contain a chain logic there. We should keep the use case calls chain in VM layer instead.